orc/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -531,6 +543,32 @@ public void nonNullWrite(int rowId, Map<K, V> map, ColumnVector output) {
     }
   }
 
+  private static class PositionDeleteWriter implements OrcRowWriter<PositionDelete<Record>> {
```

Comment:
`PositionDelete<Record>` ?  We cannot use the concrete `Record` data type here,  because when integrating the spark or flink engines,  the record to write could be spark's `InternalRow` or flink's `RowData`.  If we are only limited to write `Record` data type in this `PositionDeleteWriter`,  then the spark and flink's row type will have to be converted to `Record`.

The correct way is using a generic data type for PositionDeleteWriter,  so that we could accept `Record`, `InternalRow`, `RowData`, etc..

Reply:
The unit tests you provided is built on top of the `GenericRecord`,  so we could not find the above issue. But when we try to integrate the PositionDeleteWriter into flink or spark, the above issue will happen.